signer/src/storage/postgres.rs

```diff
@@ -92,8 +97,136 @@ impl TryFrom<&NakamotoBlock> for model::StacksBlock {
 
 impl PgStore {
     /// Connect to the Postgres database at `url`.
-    pub async fn connect(url: &str) -> Result<Self, sqlx::Error> {
-        Ok(Self(sqlx::PgPool::connect(url).await?))
+    pub async fn connect(url: &str) -> Result<Self, Error> {
+        Ok(Self(
+            sqlx::PgPool::connect(url)
+                .await
+                .map_err(Error::SqlxConnect)?,
+        ))
+    }
+
+    /// Apply the migrations to the database.
+    // Note 1: This could be generalized and moved up to the `storage` module, but
+    // left that for a future exercise if we need to support other databases.
+    // Note 2: The `sqlx` "migration" feature results in dependency conflicts
+    // with sqlite from the clarity crate.
+    // Note 3: The migration code paths have no explicit integration tests, but are
+    // implicitly tested by all integration tests using `new_test_database()`.
```

Comment:
Maybe we should link to https://github.com/stacks-network/sbtc/issues/411 so that people can read up on the underlying issue.

Also, I think the main dependency conflict happens because of the `stackslib` crate. `rusqlite` is only an optional dependency for the clarity crate (though it is brought in by default). I'm pointing this out because if we want to change this in stacks-core, that's where we should focus.

It would be nice if stacks-core provided a way to _not_ bring in sqlite for almost all of its crates. It does for stacks-common and clarity, but not for stackslib, and we need some key types from stackslib (like `StacksTransaction`, `StacksBlockId`, and `NakamotoBlock`). This would allow us to decouple any sqlite dependency in the sbtc crates from the one in stacks-core. I think stacks-core is close to being able to allow us to not pull in a sqlite dependency by being particular about which feature flags are enabled, but we need to wait for Nakamoto to push that along (I opened https://github.com/stacks-network/stacks-core/pull/5004 for that but it needs to wait).

Also, shouldn't these be `///` instead of `//`?

Reply:
Yeah it's both `stackslib` and `clarity` that cause the big problem. I spent a couple hours trying to find a way around it but ultimately gave up to not waste too much time on it.

`clarity` necessarily pulls in `rusqlite` and uses it directly (see [sqlite.rs](https://github.com/stacks-network/stacks-core/blob/master/clarity/src/vm/database/sqlite.rs)) and changing stackslib would be very difficult and time-consuming as it uses it extensively (just maybe not as obvious since they wrap `rusqlite` connections and transactions in newtypes like `DBConn`, `DBTx`,  etc. just two name a couple).

Reply:
> Also, shouldn't these be /// instead of //?

I chose `//` for the notes as they're not really code-docs in the sense that you need them as documentation when using the method, but can be interesting for anyone actually reading the code here.

I can change them to code-docs, but I think the information as rather unnecessary for a caller?

Reply:
I see. Well, maybe move it to just after the function definition? For someone reading the official crate documentation it'll be fine but when looking at the source you'll get the wrong impression.

Reply:
Changed in [e02ec59](https://github.com/stacks-network/sbtc/pull/533/commits/e02ec59a919cd54177bf6ba909cbb9bf92ef80a7)